File: YoAdopto.API/Models/Publication.cs
using System;
using System.Collections.Generic;

namespace YoAdopto.API.Models
{
    public class Publication
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int PublicationTypeId { get; set; }
        public PublicationType PublicationType { get; set; }
        public ICollection<PublicationPhoto> Photos { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public string State { get; set; }
        public string City { get; set; }
        public bool PublicationEnded { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool Active { get; set; }
    }
}

File: YoAdopto.SPA/src/app/_models/publication.ts
import { PublicationPhoto } from './publicationPhoto';

export interface Publication {
  id: number;
  title: string;
  description: string;
  publicationTypeId: number;
  username: string;
  userId: number;
  createdAt: Date;
  state: string;
  city: string;
  contactPhone: string;
  contactEmail: string;
  photoUrls?: string[];
  photos?: File[];
}
